File: hw/calle/u3/hw28PrinciplesAplication/ComputerStoreJS/controller/productController.js
//npm init -y
//npm install mongodb
//npm install express

const Product = require('../model/product');

class ProductController {
    constructor(repository, service) {
        this.repository = repository;
        this.service = service;
    }

    async handleAddProduct(req, res) {
        try {
            const { name, price } = req.body;
            const basePrice = parseFloat(price);

            const product = new Product(name, basePrice);

            const total = this.service.calculateTotal(basePrice);
            product.setTotalPrice(total);

            await this.repository.save(product);

            res.json({
                success: true,
                message: "Product saved to Atlas!",
                data: {
                    name: name,
                    total: total.toFixed(2)
                }
            });
        } catch (error) {
            res.status(500).json({ success: false, message: error.message });
        }
    }
}

module.exports = ProductController;

File: hw/hidalgo/u3/hw28PrinciplesAplication/Python/controller/product_controller.py
from model.product import Product
#py -m pip install pymongo dnspython
class ProductController:
    def __init__(self, view, repo, service):
        self.view = view
        self.repo = repo
        self.service = service
        self.view.btn_save.config(command=self.process_save)

    def process_save(self):
        try:
            name, price = self.view.get_data()
            
            total = self.service.calculate_total(price)
            
            product = Product(name, price)
            product.total_price = total
            
            self.repo.save(product)
            
            self.view.show_message(f" Total IVA: ${total:.2f}")
            self.view.clear_fields()
            
        except Exception as e:
            self.view.show_message(f"Error: {e}")